bundle/edu.gemini.sp.vcs.reg/src/main/scala/edu/gemini/sp/vcs/reg/VcsRegistrar.scala

```diff
@@ -7,7 +7,7 @@ import edu.gemini.spModel.core.Peer
  * A service that provides persistent/mutable mapping between program ids and
  * VCS server locations.
  */
-trait VcsRegistrar {
+trait VcsRegistrar extends scala.collection.mutable.Publisher[VcsRegistrationEvent] {
```

Comment:
I didn't know the `Publisher` class, what mechanism does it use to propagate events? Are there any problems due to OSGi classloaders, if you publish and listen in different bundles?